Perform hypothesis testing on the dataset above.
Draw conclusions from the results of the analysis.

### Perform hypothesis testing using COVID-19 Indonesia dataset

Dataset source: https://www.kaggle.com/datasets/hendratno/covid19-indonesia

In [28]:
# Import library used
import pandas as pd
from scipy import stats

In [2]:
# Import dataset used
covidINA = pd.read_csv("covid_19_indonesia_time_series.csv")

In [15]:
# Expand the output display to see more columns
pd.set_option('display.max_columns', None)

In [16]:
covidINA

Date Location ISO Code          Location  New Cases  New Deaths  \
0       3/1/2020             ID-JK       DKI Jakarta          2           0   
1       3/2/2020             ID-JK       DKI Jakarta          2           0   
2       3/2/2020               IDN         Indonesia          2           0   
3       3/2/2020             ID-RI              Riau          1           0   
4       3/3/2020             ID-JK       DKI Jakarta          2           0   
...          ...               ...               ...        ...         ...   
31817  9/15/2022             ID-SA    Sulawesi Utara         37           0   
31818  9/15/2022             ID-SB    Sumatera Barat         13           0   
31819  9/15/2022             ID-SS  Sumatera Selatan         16           0   
31820  9/15/2022             ID-SU    Sumatera Utara         50           0   
31821  9/16/2022               IDN         Indonesia       2358          27   

       New Recovered  New Active Cases  Total Cases  Total Deaths  \
0                  0                 2           39            20   
1                  0                 2           41            20   
2                  0                 2            2             0   
3                  0                 1            1             0   
4                  0                 2           43            20   
...              ...               ...          ...           ...   
31817              0                37        52770          1213   
31818              3                10       104640          2371   
31819              1                15        82198          3376   
31820              5                45       158866          3288   
31821           2997              -666      6405044        157876   

       Total Recovered  Total Active Cases Location Level  City or Regency  \
0                   75                 -56       Province              NaN   
1                   75                 -54       Province              NaN   
2                    0                   2        Country              NaN   
3                    1                   0       Province              NaN   
4                   75                 -52       Province              NaN   
...                ...                 ...            ...              ...   
31817            50997                 560       Province              NaN   
31818           102066                 203       Province              NaN   
31819            78510                 312       Province              NaN   
31820           154924                 654       Province              NaN   
31821          6218708               28460        Country              NaN   

               Province    Country Continent    Island  Time Zone  \
0           DKI Jakarta  Indonesia      Asia      Jawa  UTC+07:00   
1           DKI Jakarta  Indonesia      Asia      Jawa  UTC+07:00   
2                   NaN  Indonesia      Asia       NaN        NaN   
3                  Riau  Indonesia      Asia  Sumatera  UTC+07:00   
4           DKI Jakarta  Indonesia      Asia      Jawa  UTC+07:00   
...                 ...        ...       ...       ...        ...   
31817    Sulawesi Utara  Indonesia      Asia  Sulawesi  UTC+08:00   
31818    Sumatera Barat  Indonesia      Asia  Sumatera  UTC+07:00   
31819  Sumatera Selatan  Indonesia      Asia  Sumatera  UTC+07:00   
31820    Sumatera Utara  Indonesia      Asia  Sumatera  UTC+07:00   
31821               NaN  Indonesia      Asia       NaN        NaN   

               Special Status  Total Regencies  Total Cities  Total Districts  \
0      Daerah Khusus Ibu Kota                1           5.0               44   
1      Daerah Khusus Ibu Kota                1           5.0               44   
2                         NaN              416          98.0             7230   
3                         NaN               10           2.0              169   
4      Daerah Khusus Ibu Kota                1           5.0   

In [10]:
# Make new dataframe covidDate using Date for the index
covidDate = covidINA.set_index('Date').sort_values('Date')

In [17]:
covidDate

Location ISO Code             Location  New Cases  New Deaths  \
Date                                                                     
1/1/2021             ID-SA       Sulawesi Utara         53           0   
1/1/2021             ID-LA              Lampung         80           3   
1/1/2021             ID-MA               Maluku         32           1   
1/1/2021             ID-MU         Maluku Utara          2           1   
1/1/2021             ID-NB  Nusa Tenggara Barat         60           0   
...                    ...                  ...        ...         ...   
9/9/2022             ID-KT    Kalimantan Tengah         16           0   
9/9/2022             ID-KI     Kalimantan Timur         27           0   
9/9/2022             ID-KU     Kalimantan Utara          0           0   
9/9/2022             ID-JT          Jawa Tengah         82           0   
9/9/2022             ID-AC                 Aceh         10           0   

          New Recovered  New Active Cases  Total Cases  Total Deaths  \
Date                                                                   
1/1/2021             71               -18         9723           311   
1/1/2021            197              -120         6356           277   
1/1/2021             14                17         5754            73   
1/1/2021              0                 1         2773            90   
1/1/2021             20                40         5724           261   
...                 ...               ...          ...           ...   
9/9/2022             38               -22        58170          1565   
9/9/2022             49               -22       208919          5726   
9/9/2022              2                -2        45414           860   
9/9/2022            156               -74       635951         33478   
9/9/2022             13                -3        43999          2223   

          Total Recovered  Total Active Cases Location Level  City or Regency  \
Date                                                                            
1/1/2021             7150                2262       Province              NaN   
1/1/2021             4515                1564       Province              NaN   
1/1/2021             4572                1109       Province              NaN   
1/1/2021             2353                 330       Province              NaN   
1/1/2021             4575                 888       Province              NaN   
...                   ...                 ...            ...              ...   
9/9/2022            56406                 199       Province              NaN   
9/9/2022           202901                 292       Province              NaN   
9/9/2022            44529                  25       Province              NaN   
9/9/2022           601246                1227       Province              NaN   
9/9/2022            41681                  95       Province              NaN   

                     Province    Country Continent         Island  Time Zone  \
Date                                                                           
1/1/2021       Sulawesi Utara  Indonesia      Asia       Sulawesi  UTC+08:00   
1/1/2021              Lampung  Indonesia      Asia       Sumatera  UTC+07:00   
1/1/2021               Maluku  Indonesia      Asia         Maluku  UTC+09:00   
1/1/2021         Maluku Utara  Indonesia      Asia         Maluku  UTC+09:00   
1/1/2021  Nusa Tenggara Barat  Indonesia      Asia  Nusa Tenggara  UTC+08:00   
...                       ...        ...       ...            ...        ...   
9/9/2022    Kalimantan Tengah  Indonesia      Asia     Kalimantan  UTC+07:00   
9/9/2022     Kalimantan Timur  Indonesia      Asia     Kalimantan  UTC+08:00   
9/9/2022     Kalimantan Utara  Indonesia      Asia     Kalimantan  UTC+08:00   
9/9/2022          Jawa Tengah  Indonesia      Asia           Jawa  UTC+07:00   
9/9/2022                 Aceh  Indonesia      Asia       Sumatera  UTC+07:00   

         Speci

In [19]:
# Showing unique value of Location to chose province for testing
covidDate['Province'].unique()

array(['Sulawesi Utara', 'Lampung', 'Maluku', 'Maluku Utara',
       'Nusa Tenggara Barat', 'Papua', 'Papua Barat', 'Kepulauan Riau',
       'Riau', 'Sulawesi Selatan', 'Sulawesi Tengah', 'Sulawesi Tenggara',
       'Sumatera Barat', 'Sumatera Selatan', 'Sumatera Utara',
       'Sulawesi Barat', 'Kepulauan Bangka Belitung',
       'Nusa Tenggara Timur', 'Kalimantan Timur', 'Aceh', 'Bali',
       'Kalimantan Utara', 'Bengkulu', 'DKI Jakarta',
       'Daerah Istimewa Yogyakarta', 'Gorontalo', nan, 'Banten',
       'Jawa Barat', 'Jawa Tengah', 'Jawa Timur', 'Kalimantan Barat',
       'Kalimantan Selatan', 'Kalimantan Tengah', 'Jambi'], dtype=object)

Hypothesis Testing using ANOVA

we will test whether the mean of total cases of DKI Jakarta, Jawa Timur, and Sumatera Selatan are significantly different or not.

In [26]:
covidJakarta = covidDate[covidDate.Province == 'DKI Jakarta'].groupby('Date').sum(numeric_only=False)['Total Cases']
covidJawaTimur = covidDate[covidDate.Province == 'Jawa Timur'].groupby('Date').sum(numeric_only=False)['Total Cases']
covidSumateraSelatan = covidDate[covidDate.Province == 'Sumatera Selatan'].groupby('Date').sum(numeric_only=False)['Total Cases']

print("Daily Average of DKI Jakarta covid case",covidJakarta.mean().round())
print("Daily Average of Jawa Timur covid case",covidJawaTimur.mean().round())
print("Daily Average of Sumatera Selatan covid case",covidSumateraSelatan.mean().round())

Daily Average of DKI Jakarta covid case 587520.0
Daily Average of Jawa Timur covid case 263062.0
Daily Average of Sumatera Selatan covid case 38019.0


It seems that they are significantly different since `DKI Jakarta` average is `587520` case per day and the rest is between half of DKI Jakarta `38019` for `Sumatera Selatan` and `263062` for `Jawa Timur`.

In [29]:
f_stat,p_value = stats.f_oneway(covidJakarta, covidJawaTimur, covidSumateraSelatan)
print('P-value:',p_value)

P-value: 1.1294283763619467e-256


**Since the p-value is below 0.05, then we reject the Null Hypothesis.** We conclude that the difference of DKI Jakarta, Jawa Timur, and Sumatera Selatan is statistically significant.